src/tests/ftest/dfuse/pil4dfs_dcache.py

```diff
@@ -0,0 +1,360 @@
+"""
+
+import os
+import re
+
+from command_utils import ExecutableCommand
+from command_utils_base import BasicParameter
+from dfuse_test_base import DfuseTestBase
+from exception_utils import CommandFailure
+from run_utils import run_remote
+
+
+class Pil4dfsDcacheCmd(ExecutableCommand):
+    """Defines an object representing a pil4dfs_dcache unit test command."""
+
+    def __init__(self, host, path, mnt, timeout=None):
+        """Create a Pil4dfsDcacheUtestCmd object.
+
+        Args:
+            host (str): Hostname on which to remotely run the command
+            path (str): path of the DAOS install directory
+            mnt (str): path of the Dfuse mount point
+            timeout (int, optional): dir-cache timeout in seconds.
+                Default is None.
+        """
+        test_dir = os.path.join(path, "lib", "daos", "TESTING", "tests")
+        super().__init__("/run/pil4dfs_dcache/*", "pil4dfs_dcache", test_dir)
+
+        self._host = host
+
+        lib_dir = os.path.join(path, 'lib64', 'libpil4dfs.so')
+        self.env['LD_PRELOAD'] = lib_dir
+        self.env['D_DFUSE_MNT'] = mnt
+        self.env['D_LOG_MASK'] = 'DEBUG'
+        self.env['DD_SUBSYS'] = 'il'
+        self.env['DD_MASK'] = 'trace'
+        self.env['D_IL_REPORT'] = '1'
+
+        if timeout is not None:
+            self.env["D_IL_DCACHE_REC_TIMEOUT"] = timeout
+
+        self.test_id = BasicParameter(None)
+
+    @property
+    def host(self):
+        """Get the host on which to remotely run the command via run().
+
+        Returns:
+            NodeSet: remote host on which the command will run
+
+        """
+        return self._host
+
+    def _run_process(self, raise_exception=None):
+        """Run the command remotely as a foreground process.
+
+        Args:
+            raise_exception (bool, optional): whether or not to raise an exception if the command
+                fails. This overrides the self.exit_status_exception setting if defined.
+                Defaults to None.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        Returns:
+            RemoteCommandResult: a grouping of the command results from the same host with the
+                same return status
+
+        """
+        if raise_exception is None:
+            raise_exception = self.exit_status_exception
+
+        # Run pil4dfs_dcache remotely
+        result = run_remote(self.log, self._host, self.with_exports, timeout=None)
+        if raise_exception and not result.passed:
+            stdout = result.all_stdout[self.host].replace(r'\n', os.linesep)
+            stderr = result.all_stderr[self.host].replace(r'\n', os.linesep)
+            raise CommandFailure(
+                f"Error running pil4dfs_dcache_utest on: {result.failed_hosts}\n"
+                f"-- STDERR START --\n{stderr}\n-- STDERR END --\n"
+                f"-- STDOUT START --\n{stdout}\n-- STDOUT END --\n")
```

Comment:
When the command fails, `run_remote` still logs the output. It's better to not put this output in an exception message because it's very verbose when looking at test failures

Reply:
- [ ] Remove verbose error log message

Reply:
Fixed with commit eace57b3b479abd420dec07bf85cc46b4a1c9c31